packages/snap/src/stateManagement.ts

```diff
@@ -169,8 +169,8 @@ export function getHolder(
  */
 export function getZkCert(
   leafHash: string,
-  zkCerts: ZkCertRegistered[],
-): ZkCertRegistered {
+  zkCerts: ZkCertRegistered<Record<string, unknown>>[],
+): ZkCertRegistered<Record<string, unknown>> {
```

Comment:
Ok, but this function definitely must be generic. It will cover situations when caller either knows, or doesn't know the exact type he wants to get.